sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/Transforms.java

```diff
@@ -87,21 +90,12 @@ static TextDocumentBatchStatistics toBatchStatistics(RequestStatistics statistic
         return multiLanguageInputs;
     }
 
-    /**
-     * Helper method to set error details on {@link TextAnalyticsError}.
-     *
-     * @param details about specific errors that led to this reported error.
-     * @return the {@link TextAnalyticsError} returned by the SDK.
-     */
-    private static List<com.azure.ai.textanalytics.models.TextAnalyticsError> setErrors(
-        List<TextAnalyticsError> details) {
-        List<com.azure.ai.textanalytics.models.TextAnalyticsError> detailsList = new ArrayList<>();
-        for (TextAnalyticsError error : details) {
-            detailsList.add(new com.azure.ai.textanalytics.models.TextAnalyticsError(
-                ErrorCodeValue.fromString(error.getCode().toString()),
-                error.getMessage(),
-                error.getTarget(), error.getDetails() == null ? null : setErrors(error.getDetails())));
-        }
-        return detailsList;
+    static void toTextAnalyticsException(com.azure.ai.textanalytics.models.TextAnalyticsError error,
+        ClientLogger logger) {
+        String baseMessage = String.format(Locale.US, "%s: {%s}, %s",
+            "Status Code", 200, error.getMessage());
+        throw logger.logExceptionAsError(new TextAnalyticsException(baseMessage,
+            error.getCode().toString(),
```

Comment:
Is error guaranteed to have `code` or can `getCode()` result in NPE?

Reply:
Error here is guaranteed to have code and message.